src/lightning/fabric/plugins/precision/amp.py

```diff
@@ -51,7 +51,14 @@ def __init__(
 
         self.precision = precision
         if scaler is None and self.precision == "16-mixed":
-            scaler = torch.amp.GradScaler(device=device) if _TORCH_GREATER_EQUAL_2_4 else torch.cuda.amp.GradScaler()
+            scaler = (
+                torch.amp.GradScaler(device=device)
+                if _TORCH_GREATER_EQUAL_2_4
+                else getattr(
+                    torch,
+                    "cuda" if not isinstance(device, str) or device.split(":")[0] == "cpu" else device.split(":")[0],
```

Comment:
Here I think I can improve it and use torch.device.type. The problem here is that if it is a cpu backend it will lack this function, so I use cuda instead